File: language-server/LanguageServer/Protocol/RequestAdminPrivileges.cs
using MediatR;
using Newtonsoft.Json;
using OmniSharp.Extensions.JsonRpc;

namespace DanielWillett.UnturnedDataFileLspServer.Protocol;

[Parallel, Method("unturnedDataFile/requestAdminPrivileges", Direction.ServerToClient)]
public class RequestAdminPrivilegesParams : INotification
{
    [JsonProperty("message")]
    public required string Message { get; set; }

    [JsonProperty("type")]
    public required ushort Type { get; set; }
}

[Parallel, Method("unturnedDataFile/sendAdminPrivilegesResponse", Direction.ClientToServer)]
public class SendAdminPrivilegesResponseParams : INotification
{
    [JsonProperty("allowed")]
    public bool Allowed { get; set; }

    [JsonProperty("type")]
    public ushort Type { get; set; }
}

File: extensions/vscode/unturned-dat/src/jsonrpc/request-admin-privileges.ts
import { NotificationType } from 'vscode-languageclient';

export const RequestAdminPrivileges = new NotificationType<RequestAdminPrivilegesParams>("unturnedDataFile/requestAdminPrivileges");
export const SendAdminPrivilegesResponse = new NotificationType<SendAdminPrivilegesResponseParams>("unturnedDataFile/sendAdminPrivilegesResponse");

export interface RequestAdminPrivilegesParams
{
    readonly message: string,
    readonly type: number;
}

export interface SendAdminPrivilegesResponseParams
{
    readonly allowed: boolean,
    readonly type: number;
}